native/cocos/2d/renderer/Batcher2d.cpp

```diff
@@ -0,0 +1,258 @@
+#include <2d/renderer/Batcher2d.h>
+#include <cocos/base/TypeDef.h>
+#include <cocos/core/Root.h>
+#include <renderer/pipeline/Define.h>
+#include <scene/Pass.h>
+#include <deque>
+#include <iostream>
+
+namespace cc {
+Batcher2d::Batcher2d() : Batcher2d(nullptr) {
+    _device = Root::getInstance()->getDevice();
+}
+
+Batcher2d::Batcher2d(Root* root) : _drawBatchPool([]() { return ccnew scene::DrawBatch2D(); }, 10U) {
+    _root = root;
+    _device = Root::getInstance()->getDevice();
+}
+
+Batcher2d::~Batcher2d() {
+    _drawBatchPool.destroy();
```

Comment:
Should delete it too. So it is better to use `CC_SAFE_DESTROY_AND_DELETE`

Reply:
it is not a pointer